src/graph/optimizer/rule/EliminateAppendVerticesRule.cpp

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2022 vesoft inc. All rights reserved.
+//
+// This source code is licensed under Apache 2.0 License.
+
+#include "graph/optimizer/rule/EliminateAppendVerticesRule.h"
+
+#include "graph/optimizer/OptContext.h"
+#include "graph/optimizer/OptGroup.h"
+#include "graph/planner/plan/PlanNode.h"
+#include "graph/planner/plan/Query.h"
+#include "graph/util/ExpressionUtils.h"
+
+using nebula::graph::AppendVertices;
+using nebula::graph::PlanNode;
+using nebula::graph::Project;
+using nebula::graph::QueryContext;
+
+namespace nebula {
+namespace opt {
+
+// TODO If we prune properties in Optimizer Rules, we could check properties of AppendVertices,
+// and drop AppendVertices if it has no properties directly.
```

Comment:
That's right, but more conditions should be considered, such as if the var is used by other operators.
```
Match (a)-[]-(b) Return id(b);
```

Reply:
In this case, there is properties `_vid`, it's not empty too.